costmap_2d/include/costmap_2d/layer.h

```diff
@@ -118,17 +126,9 @@ class Layer
     return false;
   }
 
-  virtual bool isStaticLayer() {
-    return false;
-  }
-
-  virtual bool isObstructionLayer() {
-    return false;
-  }
-
-  virtual bool isShadowLayer() {
-    return false;
-  }
+  virtual LayerType getLayerType() {
+    return LayerType::UNKNOWN;
```

Comment:
Is it possible to create the base layer without a type?  If not we should we make this pure virtual?  If so should unknown but default.